dotcom-rendering/src/web/components/MainMedia.tsx

```diff
@@ -70,29 +70,34 @@ export const MainMedia: React.FC<{
 	host,
 	pageId,
 	webTitle,
-}) => (
-	<div
-		css={[
-			mainMedia,
-			format.display === ArticleDisplay.Immersive
-				? immersiveWrapper
-				: noGutters,
-		]}
-	>
-		{elements.map((element, index) =>
-			renderArticleElement({
-				format,
-				palette,
-				element,
-				adTargeting,
-				host,
-				index,
-				isMainMedia: true,
-				starRating,
-				hideCaption,
-				pageId,
-				webTitle,
-			}),
-		)}
-	</div>
-);
+}) => {
+	const isImmersive = format.display === ArticleDisplay.Immersive;
+	const isLiveOrDeadBlog =
+		format.design === ArticleDesign.LiveBlog ||
+		format.design === ArticleDesign.DeadBlog;
+	return (
+		<div
+			css={[
+				mainMedia,
+				isImmersive && immersiveWrapper,
+				!isImmersive && !isLiveOrDeadBlog && noGutters,
```

Comment:
I think this code has reached a point where it could benefit from a refactor. You could argue the previous addition of logic was already the start but at the time I suspect they (we, I?) decided it was okay, but I think we're at a tipping point now and given this is core code I think it makes sense to review this some more.

It seems like we're saying the margins are different in different scenarios? Standard, Immersive and Liveblog?

Reply:
Yes, it's getting a bit difficult to reason about. Would you have time to chat about a better solution?